clients/venice-admin-tool/src/main/java/com/linkedin/venice/AdminTool.java

```diff
@@ -2157,7 +2165,13 @@ private static void endMigration(CommandLine cmd) {
     ChildAwareResponse response = srcControllerClient.listChildControllers(srcClusterName);
     Map<String, ControllerClient> srcChildControllerClientMap = getControllerClientMap(srcClusterName, response);
     for (Map.Entry<String, ControllerClient> entry: srcChildControllerClientMap.entrySet()) {
-      if (entry.getValue().getStore(storeName).getStore() != null) {
+      StoreResponse childSrcStoreResponse = entry.getValue().getStore(storeName);
+      if (childSrcStoreResponse.isError() && childSrcStoreResponse.getErrorType() != ErrorType.STORE_NOT_FOUND) {
+        System.err.println(
+            "ERROR: failed to check store " + storeName + " existence in original cluster " + srcClusterName
+                + " in fabric " + entry.getKey() + " due to error: " + childSrcStoreResponse.getError());
+      }
+      if (childSrcStoreResponse.getErrorType() != ErrorType.STORE_NOT_FOUND) {
         System.err.println(
             "ERROR: store " + storeName + " still exists in source cluster " + srcClusterName + " in fabric "
                 + entry.getKey() + ". Please try again later.");
```

Comment:
I understand this is out of the scope of this PR - if something goes wrong, we shouldn't continue and reset the migration flag, right?

Reply:
No, if something goes wrong here we actually don't want to reset the migration flag for the reasons below:
1. Before calling end migration we should already verified that the most if not all traffic is shifted to destination cluster and things are working. Continue to operate this way is safer.
2. We already initiated the deletion in src cluster and the store might be in a partially deleted state and shifting the traffic back is a terrible idea.